src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -85,10 +91,40 @@ public Response forward(@NotNull @Valid RestAction ra) {
         receivedActions.increment();
 
         try {
-            restValidationClient.validate(ra.getBundle(), ra.getApplication(), ra.getEventType());
-        } catch (BadRequestException e) {
-            // The following line is required to forward the HTTP 400 error message.
-            return Response.status(BAD_REQUEST).entity(e.getMessage()).build();
+            final Response response = this.restValidationClient.validate(ra.getBundle(), ra.getApplication(), ra.getEventType());
+
+            // Close the response to avoid memory leaks.
+            if (response != null) {
+                response.close();
+            }
+        } catch (final WebApplicationException e) {
+            // Build a nice error message for the caller.
+            final Response response = e.getResponse();
+
+            final JsonObject errorResponse = new JsonObject();
+            errorResponse.put("error", response.readEntity(String.class));
+
+            Log.debugf(
+                "Unable to validate the provided rest action due to notifications-backend responding with an unexpected error. Received status code: %s, received error message: %s, received REST action in the gateway: %s",
+                response.getStatus(),
+                response.readEntity(String.class),
+                ra
+            );
+
+            // Close the response to avoid memory leaks.
+            response.close();
+
+            // Notify the caller about the error.
+            return Response
+                .status(BAD_REQUEST)
+                .entity(errorResponse)
```

Comment:
Could you please build the response entity using this?
https://github.com/RedHatInsights/notifications-gw/blob/ca380c1a89e1ec49a7e5936203a8c55b089745c5/src/main/java/com/redhat/cloud/notifications/GwResource.java#L178-L185

Reply:
Sure, done!